### PR TITLE
chore: Replace Rcpp by cpp11 

### PR DESCRIPTION
**Pull request by Antonov548**
- [x] busy_callback_helper() -> how to catch eval error and interrupt?
- [x]  result_release() -> make this work with cpp11::external_pointer
- [x]  find replacement for Rcpp::warningcall(R_NilValue, std::string("Ignoring remaining part of query: ") + tail);
- [x] figure out why cpp11::warning() crashes
- [ ] adapt astyle()
- [x] maybe remove some STRICT_R_HEADERS and R_NO_REMAP

### CODE REVIEW

**Comment by Antonov548** [comment]
@krlmlr 
Seems `cpp11` hasn't any functions to handle exceptions from R functions called in C++ via `cpp11::function`.
In Rcpp they adding `try catch` block around functions passed from R to C++ also by installing additional scripts. 

We can implement this logic in RSQLite just using R native functions or just remove catching of errors from R, at least for now.

**Comment by krlmlr** [comment]
I remember seeing a safe modifier, e.g., in https://github.com/cran/duckdb/blob/f2550069da4019ebb5f81103dd92d471c40d4d3b/inst/include/cpp11/list.hpp#L92

Would that help?

**Comment by krlmlr** [comment]
I also wonder why you commit with a different committer name and e-mail. Is this from gitpod?

**Comment by Antonov548** [comment]
> I remember seeing a safe modifier, e.g., in https://github.com/cran/duckdb/blob/f2550069da4019ebb5f81103dd92d471c40d4d3b/inst/include/cpp11/list.hpp#L92
> 
> Would that help?

Yes, I didn't think about `safe` object. I will take a look, thanks.

**Comment by krlmlr** [comment]
Thanks, weird. Can you please try this in a clean repository that doesn't have a `.gitpod.yml` ?

**Comment by Antonov548** [comment]
> Thanks, weird. Can you please try this in a clean repository that doesn't have a `.gitpod.yml` ?

Yes, I will try

File: src/rsqlite.cpp [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#include <cpp11/R.hpp>
+
```
Comment (krlmlr) [inline src/rsqlite.cpp:2]: Can we include this in `pch.h` ?

**Review by krlmlr — COMMENTED** [review]
Can we vendor the cpp11 package with `cpp11::vendor_cpp11()` and fix the tests that are now skipped?

**Comment by krlmlr** [comment]
We should un-skip the tests and fix the code instead. @Antonov548: can you please take a look?

**Comment by krlmlr** [comment]
@Antonov548: The "interrupt" test still fails legitimately. To reproduce, the package needs to be installed with `R CMD INSTALL .` from the same branch, we've omitted this today. Can you please take a look?

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**